includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -290,18 +289,22 @@ public function maybe_enforce_https_src( $src, $force_https = false ) {
 	 *
 	 * @param DOMNode|DOMElement $node The node to remove.
 	 * @param array              $args Additional args to pass to validation error callback.
-	 *
-	 * @return void
+	 * @return bool Whether the node should have been removed, that is, that the node was sanitized for validity.
 	 */
 	public function remove_invalid_child( $node, $args = array() ) {
+		$should_remove = true;
 		if ( isset( $this->args['validation_error_callback'] ) ) {
-			call_user_func( $this->args['validation_error_callback'],
+			$result = call_user_func( $this->args['validation_error_callback'],
 				array_merge( compact( 'node' ), $args )
 			);
+			if ( is_bool( $result ) ) {
+				$should_remove = $result;
+			}
 		}
-		if ( empty( $this->args['disable_invalid_removal'] ) ) {
+		if ( $should_remove ) {
 			$node->parentNode->removeChild( $node );
 		}
+		return $should_remove;
```

Comment:
In canonical AMP, whenever `$should_remove` is `false`, then this needs to trigger dirty AMP mode, where `amp` attribute is removed from the `html` element.